bin/powershell_helper.rb

```diff
@@ -0,0 +1,36 @@
+#! /usr/bin/env ruby
+#
+#   powershell_helper.rb
+#
+# DESCRIPTION:
+#   This is meant to execute powershell in the bin/powershell deirectory
+#   so we don't have to copy the code out of the GEM. This is meant only as a
+#   wrapper.
+#
+# OUTPUT:
+#   N/A
+#
+# PLATFORMS:
+#   Windows
+#
+# DEPENDENCIES:
+#   None
+#
+# USAGE:
+#   C:\opt\sensu\embedded\bin\ruby C:\opt\sensu\embedded\bin\powershell_helper <poweshell-script> ARGS
+#
+# NOTES:
+#
+# LICENSE:
+#   Copyright 2015 <dasecretzofwar@gmail.com>
+#   Released under the same terms as Sensu (the MIT license); see LICENSE for details.
+#
+
+if ARGV.empty?
+  puts 'You must specify arguments for powershell file'
+  exit 2
+end
+
+powershell_dir = File.expand_path('./powershell', File.dirname(__FILE__))
+command = "Powershell.exe -NonInteractive -NoProfile -ExecutionPolicy Bypass -NoLogo -File #{File.expand_path(ARGV.join(' '), powershell_dir)}"
```

Comment:
so I assume the whole point for this is because realistically they all need these options right? Sorry but I am not very knowledgeable about windows anymore.

Also personally I think most people would want `,` separated as this makes it harder to know the argument position.

Reply:
Long story short is that the power shell scripts reside in C:\opt\sensu\embedded\ruby\gems\2.4.0\gems\sensu_plugins_windows_version\bin\powershell so anytime that we upgrade to version of the Gem the path to the power shell changes anytime sensu changes the Ruby version the path changes so it makes it really hard to write the check definitions. The intent of the script is to simply pass whatever is in ARGS right into the power shell with no manipulation which is why I didn’t use the sensu plug-in creator this is merely a wrapper to make accessing the power shell script easier without having to to define and know the full path into them and also not having to copy the power shell script to some other Directory.